### PR TITLE
Report hot reload statistics.

### PR DESCRIPTION
**Pull request by aam**
This should help better understand how hot reload is used in real world.

### CODE REVIEW

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -663,6 +663,18 @@ class HotRunner extends ResidentRunner {
           flutterUsage.sendEvent('hot', 'reload-reject');
           return OperationResult(1, 'Reload rejected');
         } else {
+          // Collect stats that help understand scale of update for this hot reload request.
+          // For example, [syncedLibraryCount]/[finalLibraryCount] indicates how
+          // many libraries were affected by the hot reload request.
+          // Relation of [invalidatedSourcesCount] to [syncedLibraryCount] should help
+          // understand sync/transfer "overhead" of updating this number of source files.
+          final Map<String, dynamic> details = reloadReport['details'];
+          analyticsParameters['finalLibraryCount'] = "${details['finalLibraryCount']}";
```
Comment (DanTup) [inline packages/flutter_tools/lib/src/run_hot.dart]: I don't know a lot about these analytics (or the `usage` package) but the one I added (`kEventReloadReasonParameterName`) has a key of `cd5` which maps to a custom dimension in the analytics account (which someone with access had to create).

I assumed all the fields had to be known keys (eg. from [here](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters)) - is this not the case?

Comment (aam) [reply packages/flutter_tools/lib/src/run_hot.dart]: Ah! Okay. Reserved cd6-11 for this metrics.

Comment (aam) [reply packages/flutter_tools/lib/src/run_hot.dart]: cc @jayoung-lee

**Review by a-siva — APPROVED** [review]

**Review by DanTup — APPROVED** [review]
The custom dimensions may need creating in Analytics before the data will show up, but otherwise, LGTM!

**Pull request merged**